### PR TITLE
feature (ami upgrade): split ami upgrade to two jobs

### PR DESCRIPTION
**Pull request by amoskong**
Currently 2021.1 in base_version switched to Ubuntu AMI, we should pass
different kind of new_scylla_repo to the subtests.

This patch split the tests to two jobs. I will create a new jenkins job,
and asked rel-eng to trigger it with Ubuntu AMI.

Signed-off-by: Amos Kong <amos@scylladb.com>

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by amoskong** [comment]
byo job: https://jenkins.scylladb.com/job/enterprise-2021.1/job/rolling-upgrade/job/rolling-upgrade-ubuntu-ami-test/1/

**Review by fgelcer — CHANGES_REQUESTED** [review]
also, why not to move this job to be ubuntu only, and create a new `rolling-upgrade-centos-ami` as the ubuntu should last longer, and once not supported anymore the centos job, we can remove it

File: jenkins-pipelines/rolling-upgrade-ubuntu-ami.jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!groovy
+
+// trick from https://github.com/jenkinsci/workflow-cps-global-lib-plugin/pull/43
+def lib = library identifier: 'sct@snapshot', retriever: legacySCM(scm)
+
+rollingUpgradePipeline(
+    backend: 'aws',
+    base_versions: ['2021.1'],
+    linux_distro: 'centos',
```
[the remark this answers is not in the document]
Comment (amoskong) [reply jenkins-pipelines/rolling-upgrade-ubuntu-ami.jenkinsfile:9]: Good catch, actually we used linux_distro: 'centos' for all jobs that use ubuntu AMI.
It works but we need to fix it in SCT. Maybe we can switch to general `scyllaadm`?

/Cc @bentsi @roydahan

[the remark this answers is not in the document]
Comment (amoskong) [reply jenkins-pipelines/rolling-upgrade-ubuntu-ami.jenkinsfile]: scylla-cluster-tests]$ git grep ami_db_scylla_user
defaults/aws_config.yaml:ami_db_scylla_user: 'centos'
test-cases/artifacts/amazon2.yaml:ami_db_scylla_user: 'ec2-user'
test-cases/artifacts/oel76.yaml:ami_db_scylla_user: 'clckwrk'
test-cases/artifacts/oel81.yaml:ami_db_scylla_user: 'ec2-user'

**Comment by fgelcer** [comment]
> New byo:
> 
> * https://jenkins.scylladb.com/job/enterprise-2021.1/job/Reproducer/job/rolling-upgrade-ubuntu-ami-test-debug/1/console
>   
>   * ignore the result, job failed for other reason.
> 
> Verified the region fix: both builder and instances use eu-north-1 (not default).

@amoskong , your BYO job failed

**Comment by amoskong** [comment]
> > New byo:
> > 
> > * https://jenkins.scylladb.com/job/enterprise-2021.1/job/Reproducer/job/rolling-upgrade-ubuntu-ami-test-debug/1/console
> >   
> >   * ignore the result, job failed for other reason.

> > Verified the region fix: both builder and instances use eu-north-1 (not default).
> 
> @amoskong , your BYO job failed

The byo job failed for other unrelated problem. I have other PR to fix it.
I'm running another BYO with that fix. will update here when it succeeds.

File: jenkins-pipelines/rolling-upgrade-ubuntu-ami.jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,17 @@
+#!groovy
+
+// trick from https://github.com/jenkinsci/workflow-cps-global-lib-plugin/pull/43
+def lib = library identifier: 'sct@snapshot', retriever: legacySCM(scm)
+
+rollingUpgradePipeline(
+    backend: 'aws',
+    base_versions: ['2021.1'],
```
[the remark this answers is not in the document]
Comment (amoskong) [reply jenkins-pipelines/rolling-upgrade-ubuntu-ami.jenkinsfile:8]: 2021.1 will be filtered out in upgrade pipeline of sct master.
We used SCT master to test enterprise, so we keep 2021.1 in sct master.

File: jenkins-pipelines/rolling-upgrade-ami.jenkinsfile [hunk 3]
```diff
@@ -5,7 +5,7 @@ def lib = library identifier: 'sct@snapshot', retriever: legacySCM(scm)
 
 rollingUpgradePipeline(
     backend: 'aws',
-    base_versions: ['2020.1', '4.3', '2021.1'],
+    base_versions: ['2020.1', '4.3'],
```
[the remark this answers is not in the document]
Comment (amoskong) [reply jenkins-pipelines/rolling-upgrade-ami.jenkinsfile:8]: > In master we shouldn't have 2020.1 here.
> Only "latest OSS".

same here:

```
2021.1 will be filtered out in upgrade pipeline of sct master.
We used SCT master to test enterprise, so we keep 2021.1 in sct master.
```

> We should find a way to use here "latest" so we can use it without updating it every release.

We can get the latest version by checking repo filename in 
- http://downloads.scylladb.com/rpm/centos/
- http://downloads.scylladb.com/deb/ubuntu/
- http://downloads.scylladb.com/deb/debian/

**Review by roydahan — APPROVED** [review]

**Comment by roydahan** [comment]
Backported, please make sure the jobs runs with official branch and pass.

File: vars/rollingUpgradePipeline.groovy [hunk 4]
```diff
@@ -84,6 +84,10 @@ def call(Map pipelineParams) {
 
                                                         export SCT_CLUSTER_BACKEND=${params.backend}
 
+                                                        if [[ -n "${params.aws_region ? params.aws_region : ''}" ]] ; then
+                                                            export SCT_REGION_NAME=${aws_region}
```
Comment (vponomaryov) [inline vars/rollingUpgradePipeline.groovy:88]: Shouldn't it be "params.aws_region" ? I don't see definition of the "aws_region" variable.

Comment (fgelcer) [reply vars/rollingUpgradePipeline.groovy:88]: i think not, as if we don't have `params.aws_region` or if it is an empty string, we take the value from the pipeline field called `aws_region` here:

```
string(defaultValue: "${pipelineParams.get('aws_region', 'eu-west-1')}",
               description: 'Supported: us-east-1|eu-west-1|eu-west-2|eu-north-1|random (randomly select region)',
               name: 'aws_region')
```

**Pull request merged**